subprojects/architecture-test/src/changes/archunit_store/injected-services-should-have-service-scope-applied.txt

```diff
@@ -0,0 +1,96 @@
+Class <org.gradle.configuration.project.ProjectEvaluator> is not annotated with @ServiceScope in (ProjectEvaluator.java:0)
+Class <org.gradle.deployment.internal.DeploymentRegistry> is not annotated with @ServiceScope in (DeploymentRegistry.java:0)
+Class <org.gradle.execution.selection.BuildTaskSelector$BuildSpecificSelector> is not annotated with @ServiceScope in (BuildTaskSelector.java:0)
+Class <org.gradle.execution.taskgraph.TaskExecutionGraphInternal> is not annotated with @ServiceScope in (TaskExecutionGraphInternal.java:0)
+Class <org.gradle.groovy.scripts.internal.ScriptCompilationHandler> is not annotated with @ServiceScope in (ScriptCompilationHandler.java:0)
+Class <org.gradle.initialization.BuildClientMetaData> is not annotated with @ServiceScope in (BuildClientMetaData.java:0)
+Class <org.gradle.initialization.DependenciesAccessors> is not annotated with @ServiceScope in (DependenciesAccessors.java:0)
+Class <org.gradle.initialization.ProjectDescriptorRegistry> is not annotated with @ServiceScope in (ProjectDescriptorRegistry.java:0)
+Class <org.gradle.internal.actor.ActorFactory> is not annotated with @ServiceScope in (ActorFactory.java:0)
+Class <org.gradle.internal.build.PublicBuildPath> is not annotated with @ServiceScope in (PublicBuildPath.java:0)
+Class <org.gradle.internal.file.PathToFileResolver> is not annotated with @ServiceScope in (PathToFileResolver.java:0)
+Class <org.gradle.internal.installation.CurrentGradleInstallation> is not annotated with @ServiceScope in (CurrentGradleInstallation.java:0)
+Class <org.gradle.internal.instantiation.InstantiatorFactory> is not annotated with @ServiceScope in (InstantiatorFactory.java:0)
+Class <org.gradle.internal.logging.LoggingManagerInternal> is not annotated with @ServiceScope in (LoggingManagerInternal.java:0)
+Class <org.gradle.internal.logging.progress.ProgressLoggerFactory> is not annotated with @ServiceScope in (ProgressLoggerFactory.java:0)
+Class <org.gradle.internal.logging.text.StyledTextOutputFactory> is not annotated with @ServiceScope in (StyledTextOutputFactory.java:0)
+Class <org.gradle.internal.nativeintegration.filesystem.FileSystem> is not annotated with @ServiceScope in (FileSystem.java:0)
+Class <org.gradle.internal.nativeintegration.network.HostnameLookup> is not annotated with @ServiceScope in (HostnameLookup.java:0)
+Class <org.gradle.internal.operations.logging.BuildOperationLoggerFactory> is not annotated with @ServiceScope in (BuildOperationLoggerFactory.java:0)
+Class <org.gradle.internal.reflect.Instantiator> is not annotated with @ServiceScope in (Instantiator.java:0)
+Class <org.gradle.internal.resource.TextUriResourceLoader$Factory> is not annotated with @ServiceScope in (TextUriResourceLoader.java:0)
+Class <org.gradle.internal.service.ServiceRegistry> is not annotated with @ServiceScope in (ServiceRegistry.java:0)
```

Comment:
🤔 I think we should exclude `ServiceRegistry` (and maybe a few other magically available things) from the check without using the freeze file.
Maybe in the short term we also should have an annotation `@NoServiceScope("<reason>")` that allows to specify on a class why it doesn't need a service scope although it can be injected.

Reply:
If you have a list of things here, which you think special, please write it down

Reply:
Updated the rules to mark "special" classes